src/Program.ts

```diff
@@ -74,12 +74,13 @@ export class Program {
         //normalize the root dir path
         this.options.rootDir = util.getRootDir(this.options);
 
-        this.createGlobalScope();
+        this.globalScope = this.createGlobalScope();
```

Comment:
What was the purpose of changing this to an assignment? The global scope was created _in_ `createGlobalScope`.

Reply:
This was to make it easier to convince TS that this was indeed created in the constructor. As an alternative I could declare the scope property on the class as `public globalScope: Scope = undefined as any`, which might be less disruptive.

Reply:
Updated to `public globalScope: Scope  = undefined as any`